### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by RustyBamboo**

### CODE REVIEW

File: client/screens/screengame.cpp [hunk 1]
```diff
@@ -1,42 +1,131 @@
...
-        mob->tick();
-        mob->draw(window);
-
-        worldMap.tick();
-        worldMap.draw(window);
-        window.display();
-    }
-
-    return -1;
+	sf::Event Event;
```
Comment (RustyBamboo) [inline client/screens/screengame.cpp:121]: Needs to have while(running) loop

File: client/screens/screengame.cpp [hunk 2]
```diff
@@ -1,42 +1,131 @@
...
+		else if (Event.type == sf::Event::KeyPressed) {
+			switch (Event.key.code) {
+				case sf::Keyboard::Escape:
+					return (0); //Goes to Pause Menu
+			}
+		}
+		else {
+			doTick();
+			worldMap.draw(window);
+		}
```
Comment (RustyBamboo) [inline client/screens/screengame.cpp]: draw and tick should not be in event loop

Comment (RustyBamboo) [reply client/screens/screengame.cpp]: Also, need window.clear() and window.display()

File: client/screens/screengame.cpp [hunk 3]
```diff
@@ -1,42 +1,131 @@
...
+				{
+					printf("Update UPDATE_POLYGON #%u ID=%u\n",i,update.id);
+					Polygon* p = (Polygon*) worldMap.getEntity(update.id);
+					*p << packet;
+				}
+				break;
+			case UpdatePacket::NEW_POLYGON:
+				{
+					printf("Update NEW_POLYGON #%u ID=%u\n",i,update.id);
+					Polygon *p = new Polygon();
```
Comment (RustyBamboo) [inline client/screens/screengame.cpp]: Is this pointer deleted after out-of-scope?

Comment (RustyBamboo) [reply client/screens/screengame.cpp]: Probably not...

**Comment by RustyBamboo** [comment]
... UDP is where it's at...http://gafferongames.com/networking-for-game-programmers/

**Comment by kev-the-dev** [comment]
cool

**Pull request closed without merging**